iothub/service/src/RegistryManager.cs

```diff
@@ -62,14 +62,20 @@ public static RegistryManager CreateFromConnectionString(string connectionString
         /// <param name="hostName">IoT hub host name.</param>
         /// <param name="credential">Azure Active Directory credentials to authenticate with IoT hub. See <see cref="TokenCredential"/></param>
         /// <param name="transportSettings">The HTTP transport settings.</param>
+        /// <param name="options">Options that allow configuration of the RegistryManager instance during initialization.</param>
         /// <returns>An instance of <see cref="RegistryManager"/>.</returns>
         /// <remarks>
         /// For more information on configuring IoT hub with Azure Active Directory, see <see href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-dev-guide-azure-ad-rbac"/>
+        /// This constructor sets the default for <see cref="RegistryManagerOptions.TokenCredentialAuthenticationScopes"/> to
+        /// <see cref="IotHubAuthenticationScopes.DefaultAuthenticationScopes"/>, which is used for any public or private cloud other than Azure US Government cloud.
+        /// For Azure US Government cloud users, set the <see cref="RegistryManagerOptions.TokenCredentialAuthenticationScopes"/>
+        /// to <see cref="IotHubAuthenticationScopes.AzureGovernmentAuthenticationScopes"/>.
         /// </remarks>
         public static RegistryManager Create(
             string hostName,
             TokenCredential credential,
-            HttpTransportSettings transportSettings = default)
+            HttpTransportSettings transportSettings = default,
+            RegistryManagerOptions options = default)
```

Comment:
Having two options classes here and elsewhere feels a bit odd to me. What about having RegistryManagerOptions contain all the same settings as HttpTransportSettings and then add the new authentication scopes setting.

```csharp
// leave old constructor alone, maybe deprecate it
public static RegistryManager Create(
    string hostName,
    TokenCredential credential,
    HttpTransportSettings transportSettings = default)

// newly added constructor
public static RegistryManager Create(
    string hostName,
    TokenCredential credential,
    RegistryManagerOptions options = default)
```

Reply:
It is an option but it would introduce another overload, which leads to a user trying to think which to use. Everyone is already used to using TransportSettings and moving the same things to another clientOptions class seemed confusing to me. TransportSettings seemed to too specific so I created the other options class instead of adding it there.

Reply:
Rather than use ctor overloads, should we consider relying on `readonly` property initialization?

Reply:
The current design does what you are suggesting I suppose. Will talk offline.